### DISCUSSION TITLE
SageTV Locator Service

### DISCUSSION THREAD

**Discussion by Narflex** [OpenSageTV/Community#6] [General]
Placeholder for modifying configurations to use the SageTV locator service after the June 1st, 2023 shutdown. (I'm going to link to this from the announcement in the archived forums)

**Comment by sic0048** [comment]
A self hosted VPN service is probably the best way to connect to a remote SageTV system anyway.

**Reply by wayner9** [reply]
And stuff like Wireguard or OpenVPN is pretty easy to set up these days.  Especially Wireguard.

**Comment by Sagetvmaster** [comment]
Wasn't there a point where Narflex was going to find an alternative way to do this, or is current update that it will be shutdown in a few days?

**Comment by Narflex** [comment]
Yeah, I am going to change the SageTV server code so that you can specify an alternate locator server. But we haven't set that up yet...it's very likely there will be a gap between the shutdown and the new server being available. But I will do it because I actually want it for my own purposes as well. :)

**Reply by Sagetvmaster** [reply]
Cool beans, thanks.  I’ve been using DNS sites for years now when traveling and testing.

**Comment by Narflex** [comment]
I pushed an update to the SageTV code that now enables configuring the locator server location:

https://github.com/google/sagetv/commit/bd0410e67ea5fed38dc828ba8d450e0eb9090267

That means there will need to be a new JAR for servers to use and then also for placeshifter clients. The Android client will also need to be updated to handle this too.  The new domain name for the locator server is locator.mu3d.com (the default w/ the updated code).

If you were using locator.sagetv.com/locator.php to redirect to your home webserver...you can still use that at the locator.mu3d.com location (that will then also provide IP address resolution for a locator ID to deal with clients that are not updated for the new server).

**Reply by Sagetvmaster** [reply]
Hey there Jeff.  On a different note - any idea why the old, static site cannot be searched within?  I seem to remember testing that after the shutdown and thought it worked.  Now, there seem to be php and certificate problems.
Thoughts?
Thanks

**Reply by Narflex** [reply]
When I posted on the forums that it was shutdown...it actually wasn't shutdown yet. Once you started seeing the certificate problems and searching not working...that's when you were actually hitting the new static site. And it's not possible for searching to work anymore...because the site is static and has no backend functionality now.

The certificate problems should sort themselves out within the next couple of weeks. I did check internally w/ some other people to make sure I had SSL configured correctly and they indicated it is...it's just the time it takes for changes like this to roll out is about 4 weeks, so it's not fully rolled out yet.

**Reply by eelpout** [reply]
Hey Jeff. The html pages exist. Is it possible for the site to be reindexed by Google so that results point to those pages rather than the scripted PHP now that it's static (because those results are broken too)? Or might that just happen eventually anyway once the certificate issue is resolved? Thanks.

**Reply by Narflex** [reply]
I think it might happen once the SSL certificate problem is resolved....so I'm waiting until that propagates to see if I need to take further action on that or not. But I definitely intend to have it all re-indexed once it is possible to do so.

**Reply by Sagetvmaster** [reply]
Ok, then regarding the WinHTTrack site-rip I did - that also worked for searches, but yes, that was early June I tested it.  I'm guessing even the downloaded site that's on the drive somehow is referencing the old online site, as searching doesn't work on the HTTrack version either.  What am I not understanding about the HTTrack concept?  Thanks for the quick replay :)

**Reply by Narflex** [reply]
What likely happened is you clicked on a link in the site rip that took you to the real website and then the search was performed there. There's no way searching will work in an offline site copy (unless you're using grep locally or some other kind of local search mechanism).

**Reply by Sagetvmaster** [reply]
Hmmm, OK.  I just assumed since HTTrack created the image from start to finish, it would be looking within the drive itself for all files.  I didn't realize the links would be addressed to an external location, but I guess that makes sense.  
There's no way for a program to auto scan the entire dump and re-link everything internally I guess?

**Reply by Narflex** [reply]
Everything is re-linked internally...but pages that it did not download will revert to using their existing links to no longer existing pages.

**Reply by Sagetvmaster** [reply]
I used Total Commander’s ‘find text’ search feature, which while a bit clugey, has at least found several html pages with the words I’m looking for.  Surprised it didn’t download all the pages in 7gb.

**Comment by Sagetvmaster** [comment]
Has anyone taken time to work with this new update from Narflex regarding the locator service?  I have not, but my brain siezes when lines of code appear, so I'm not the best reference here...

**Comment by Sagetvmaster** [comment]
Nobody?  Anyone using what Narflex set up?

**Comment by Sagetvmaster** [comment]
Random chance someone is reading this - did anyone ever implement these 'locator' changes from 2.5 years ago to use Placeshifter?
